runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -97,6 +97,7 @@ function TSHighlighter.new(tree, opts)
   if vim.g.syntax_on ~= 1 then
     vim.api.nvim_command('runtime! syntax/synload.vim')
   end
+  vim.bo[self.bufnr].spelloptions = 'noplainbuffer'
```

Comment:
This change seems to override a user's ability to include `camel` inside their spell options, think it should be an `append` instead